manage_app_pool - added app pool identity capability

lcoulson opened the pull request:
new params kept optional as undef to allow for backwards compatibility,
logic in class will handle undef

Caller MUST specify $apppoolusername $apppoolpassword if using
"SpecificUser" identity type

igalic commented:
@lcoulson thank you very much for this patch!

This will need docs. and [spec](https://github.com/voxpupuli/puppet-iis/blob/master/spec/defines/manage_app_pool_spec.rb) tests.

lcoulson commented:
oh stars, ill see what I can do. 

one more to come, fixed issues I'd had with using future parser features....  for next one I need to validate regex of each array member from input - I might be able to flatten it and verify that directly, i'd also been using lambdas to process each array member...

igalic commented:
\o/

lcoulson commented:
@igalic @juniorsysadmin got up todate with upstream changes, fixed variable naming camelCase (grumble grumble), added rspec tests, added documentation examples
